### ISSUE TITLE
Bug Report - Cannot move Azure policy assignment from subscription level to management group level 

### ISSUE THREAD

**Issue by MM-MikeM** [Azure/AzOps#587]
**Describe the bug**
We would like to move a policy assignment from the subscription level to the management group level.  The definition location of the policy is at the management group level.  Every time the steps below are performed and a pull is initiated the policy assignment is changed back to its initial state.

**Steps to reproduce**

1. Change scope in policy assignment json file.
2. Move the policy assignment json file up to the management group folder.

**Screenshots**

**Comment by Jefajers** [comment]
Hi @MM-MikeM, I understand your desired outcome and plan of action.

So to move a policy assignment from subscription ‘x’ to management group ‘y’ you have updated the scope value to reflect the desired management group and moved the file itself to that management group.

**Suggested approach**
A change in the steps here would help with the outcome you are looking for.

Instead of moving the file, create a new file (empty) at the management group scope and copy the content from source policy assignment and paste into new empty file, update the scope and delete the policy file at subscription scope.

Performing the steps above will ensure the deletion is handled and the new policy assignment gets deployed.

**What is happening?**
1. Create branch
2. Find file for the existing policy assignment at subscription scope
3. Update scope section in file to point towards new mgmt scope
4. Move file
6. Create Pull Request
7. AzOps runs validation of the suggested change
8. Once PR is merged to main
9. AzOps runs push job and creates the new policy assignment at desired mgmt scope
10. AzOps runs pull job and finds that there is (still) a policy assignment at subscription scope ‘x’ so it brings that over as code into main.

**Why?**
The reason we end up with both a file at the subscription scope ‘x’ and management group scope ‘y’ is because we “moved” a file in Git. Git is smart and does not report that the file is deleted but instead it is renamed. The trigger for AzOps to perform deletions of policy assignments are triggered when the git diff results match a deletion.

So in short the current “behavior” is excepted based on the current pipeline configuration and AzOps module configuration.

**Comment by uday31in** [comment]
I wonder it it make sense now to separate rename operation here to handle this scenario more effectively.

https://github.com/Azure/AzOps/blob/ac67ba5f784a178dbba1c2bf4444251c89fb823e/src/functions/Invoke-AzOpsPush.ps1#L178

**Comment by Jefajers** [comment]
Agree that if we deterministically can catch renames and from statements the move experience could be improved, by adding logic to `Invoke-AzOpsPush.ps1` and `validate-deploy.yml.` Labeling this issue as a future enhancement to investigate further.

https://github.com/Azure/AzOps-Accelerator/blob/2a75754ff908ed4997e39862d092a43dd084af71/.github/actions/validate-deploy/action.yml#L27
